projects/OG-Analytics/src/main/java/com/opengamma/analytics/financial/interestrate/future/provider/BondFutureOptionMarginSecurityBlackSmileMethod.java

```diff
@@ -0,0 +1,299 @@
+   * @return The instance.
+   */
+  public static BondFutureOptionMarginSecurityBlackSmileMethod getInstance() {
+    return INSTANCE;
+  }
+
+  /**
+   * The Black function used in the pricing.
+   */
+  private static final BlackPriceFunction BLACK_FUNCTION = new BlackPriceFunction();
+
+  /**
+   * The method used to compute the future price. 
+   */
+  private static final BondFuturesSecurityDiscountingMethod METHOD_FUTURE = BondFuturesSecurityDiscountingMethod
+      .getInstance();
+
+  /**
+   * Computes the option security price from future price.
+   * @param security The future option security, not null
+   * @param black The curve and Black volatility data, not null
+   * @param price The underlying futures price.
+   * @return The security price.
+   */
+  public double priceFromUnderlyingPrice(BondFuturesOptionMarginSecurity security,
+      BlackBondFuturesProviderInterface black, double price) {
+    ArgumentChecker.notNull(security, "security");
+    ArgumentChecker.notNull(black, "Black data");
+    final double strike = security.getStrike();
+    final EuropeanVanillaOption option = new EuropeanVanillaOption(strike, security.getExpirationTime(),
+        security.isCall());
+    final double delay = security.getUnderlyingFuture().getNoticeLastTime() - security.getExpirationTime();
+    final double volatility = black.getVolatility(security.getExpirationTime(), delay, strike, price);
+    final BlackFunctionData dataBlack = new BlackFunctionData(price, 1.0, volatility);
+    final double priceSecurity = BLACK_FUNCTION.getPriceFunction(option).evaluate(dataBlack);
+    return priceSecurity;
+  }
+
+  /**
+   * Computes the option security price without future price.
+   * @param security The future option security, not null
+   * @param black The curve and Black volatility data, not null
+   * @return The security price.
+   */
+  public double price(BondFuturesOptionMarginSecurity security, BlackBondFuturesProviderInterface black) {
+    ArgumentChecker.notNull(security, "security");
+    ArgumentChecker.notNull(black, "Black data");
+    double priceFutures = METHOD_FUTURE.price(security.getUnderlyingFuture(), black.getIssuerProvider());
+    return priceFromUnderlyingPrice(security, black, priceFutures);
+  }
+
+  /**
+   * Computes the option security price curve sensitivity. 
+   * It is supposed that for a given strike the volatility does not change with the curves.
+   * The option price and its derivative wrt the futures price is computed using the futures price. 
+   * The derivatives of the futures price with respect to the curves are computed using the curves.
+   * @param security The future option security, not null
+   * @param black The curve and Black volatility data, not null
+   * @param price The underlying futures price.
+   * @return The security price curve sensitivity.
+   */
+  public MulticurveSensitivity priceCurveSensitivityFromPrice(final BondFuturesOptionMarginSecurity security,
```

Comment:
For consistency with the price method, use "priceCurveSensitivityFromUnderlyingPrice"